doc/releases/migration-guide-4.2.rst

```diff
@@ -46,3 +46,13 @@ Modules
 
 Architectures
 *************
+
+Ethernet
+********
+
+* STM32 HAL Ethernet Changes:
+
+  * Removed the Kconfig option ``ETH_STM32_HAL_MII``:
+    * This option was used to select the MII physical interface instead of RMII.
+    * The selection of the PHY interface type should now be handled through
+      the ``phy-connection-type`` property in the device tree.
```

Comment:
this belongs to ``Device Drivers and Devicetree``, take a look at ``migration-guide-4.1.rst`` for reference.

can also be a bit shorter